Updates

kevincantu opened the pull request:
An attempt to combine https://github.com/PaloAltoNetworks/elemental/pull/137, https://github.com/PaloAltoNetworks/elemental/pull/138, and some other updates

james-otten-pan committed:
Remove cov

james-otten-pan committed:
Update build-go.yaml

kevincantu committed:
Pin action versions

kevincantu committed:
Try using golangci-lint v1.64.8 (pre 2.0)

kevincantu committed:
Try using golangci-lint v1.54.2 (earlier, pre 2.0), and update deps

kevincantu committed:
Configure linters, particularly revive in golangci-lint v2 for Go 1.24

kevincantu committed:
Linting

kevincantu committed:
Linter configs

kevincantu committed:
No really v2

kevincantu committed:
Show version

kevincantu commented:
Hmm, same versions locally, different linter behavior.

kevincantu committed:
Linting

kevincantu committed:
Try disabling a few more cases

kevincantu committed:
Linting

kevincantu committed:
Linting

kevincantu commented on .golangci.yaml:
```diff
@@ -0,0 +1,82 @@
+        - name: confusing-results
+          disabled: true
+
+        # kevin recommends not disabling these
+        - name: unchecked-type-assertion
+          disabled: true
```
@brianonn can you take a look at the linter results when we don't disable this?
https://github.com/PaloAltoNetworks/elemental/actions/runs/14814866494/job/41594086119

james-otten-pan reviewed (dismissed):
This accomplishes what I was hoping to accomplish in this repo.

brianonn committed:
run with linter for unchecked-type-assertion enabled

brianonn committed:
fix regression error with map equality tests in push_config

Test code uses smarty Convey to check for two maps not equal, but
internally smarty uses reflect.DeepEqual which returns true, but the
test code expects it to return false, even though the map[] contents are
the same (deeply equal) but the maps are uniquely different in memory.
Maybe in the past this was returned as not equal, but today they return
as equal.  So I changed it to use reflect.ValueOf to validate the actual
objects are not the same.

it fixes the following error
* /home/runner/work/elemental/elemental/push_config_test.go
Line 508:
Expected     'map[i1:[create delete] i2:[create delete]]'
to NOT equal 'map[i1:[create delete] i2:[create delete]]'
(but it did)!

brianonn commented:
@kevincantu I pushed two commits to 
1. re-enable the `unchecked-type-assertion` linter and fix the type assertions it found; and 
2. fixed a smarty assertion that used to work but stopped working now . It seems like the way smarty handles map equality changed either between Go 1.23 and 1.24 (if reflect.DeepEqual changed for maps) or perhaps an update to smarty assertions changed something.

kevincantu commented:
Thanks @brianonn and @james-otten-pan!
